### PR TITLE
test(hx-pill test cases): surf2015

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Unit Test Case `<hx-pill>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2015

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [x] If you changed/added functionality, did you update the demo page and documentation?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-pill/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { fixture, oneEvent, expect } from '@open-wc/testing';
+
+/**
+ * <hx-pill> component tests
+ *
+ * @type HXPillEvent
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:6]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ */
+describe('<hx-pill> component tests', () => {
+    const template = '<hx-pill>';
+
+    describe('test instantiate element', () => {
+        it('should be instantiated with hx-defined attribute', async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:14]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            const attr = component.hasAttribute('hx-defined');
+
+            expect(attr).to.be.true;
+        });
+
+        it('should not be hidden', async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:21]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            const prop = component.hidden;
+
+            expect(prop).to.be.false;
+        });
+
+        it(`the rendered Light DOM should NOT equal simple template ${template}`, async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:28]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+            expect(component).lightDom.to.not.equal(template);
+        });
+
+    });
+
+    describe('test Shadow DOM', () => {
+        describe('verify render', () => {
+            it('should have a static Shadow DOM', async function () {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:38]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:45]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
+            });
+
+            it('should have a single <slot>', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:52]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(1);
+            });
+
+            it('should have an unnamed <slot>', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:61]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+                }
+            });
+        });
+
+        describe('verify Shadow DOM markup', () => {
+
+            it('markup should contain a #hxDismiss button', async () => {
+                const elSelector = 'button#hxDismiss';
+                const id = 'hxDismiss';
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:77]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxDismiss button with a times type <hx-icon>', async () => {
+                const elSelector = 'button#hxDismiss > span > hx-icon';
+                const iconType = 'times';
+                const component = /** @type { HXPillEvent } */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:88]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:91]: Please change variable to `queryType`.

File: src/elements/hx-pill/index.spec.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+    describe('test getters and setters', () => {
+        it('should be able to persist', async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
+            component.persist = true;
+            const attr = component.hasAttribute('persist');
+
+            expect(attr).to.be.true;
+        });     
+    });
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:106]: Please add a test for the `click` event.

https://github.com/HelixDesignSystem/helix-ui/blob/master/src/elements/hx-pill/index.js#L35

File: src/elements/hx-pill/index.spec.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+                const queryId = query.type;
+
+                expect(queryId).to.equal(iconType);
+            });
+        });
+    });
+
+    describe('test getters and setters', () => {
+        it('should be able to persist', async () => {
+            const component = /** @type {HXPillEvent} */ await fixture(template);
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:100]: Please change to `HXPillElement`.

File: src/elements/hx-pill/index.spec.js [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+                const shadow = component.shadowRoot.innerHTML;
+
+                expect(component).shadowDom.to.equal(shadow);
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXPillEvent } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal("open");
```
Comment (100stacks) [inline src/elements/hx-pill/index.spec.js:48]: Please use single quotes here - `'open'`.

File: src/elements/hx-pill/index.spec.js (hunk 1 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 2 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 3 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 4 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 5 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 6 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 8 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 9 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 10 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 11 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 12 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js:106]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 13 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-pill/index.spec.js (hunk 14 above)
Comment (lalithkarikelli) [reply src/elements/hx-pill/index.spec.js]: Changes have been done, please review.

**Pull request merged**